tools/azsdk-cli/Azure.Sdk.Tools.Cli.Tests/Tools/Generators/ReadMeGeneratorToolTests.cs

```diff
@@ -128,45 +141,6 @@ public async Task TestBadReadmeContent(string readmeContent, string expectedFeed
             }
         }
 
-        /// <summary>
-        /// Creates a service provider in the same way as the normal program. Can be used to instantiate real clients.
-        /// </summary>
-        /// <param name="customizeServices">Optional Action you can use to register your own client instances</param>
-        /// <returns></returns>
-        private static (ServiceProvider, MockOutputHelper) CreateServiceProvider(Action<ServiceCollection>? customizeServices = default)
```

Comment:
I had this in here specifically to use the service collection. I'm not sure why we should remove it if all we're trying to do is use a fake logger.

Can we split this out from the changes for this particular PR?